rq/queue.py

```diff
@@ -32,33 +32,35 @@ class Queue(object):
     redis_queues_keys = 'rq:queues'
 
     @classmethod
-    def all(cls, connection=None):
+    def all(cls, connection=None, namespace=None):
         """Returns an iterable of all Queues.
         """
         connection = resolve_connection(connection)
+        redis_queues_keys = build_key(cls.redis_queues_keys, namespace)
 
         def to_queue(queue_key):
-            return cls.from_queue_key(as_text(queue_key),
-                                      connection=connection)
-        return [to_queue(rq_key) for rq_key in connection.smembers(cls.redis_queues_keys) if rq_key]
+            return cls.from_queue_key(as_text(queue_key), connection=connection, namespace=namespace)
+        return [to_queue(rq_key) for rq_key in connection.smembers(redis_queues_keys) if rq_key]
 
     @classmethod
-    def from_queue_key(cls, queue_key, connection=None):
+    def from_queue_key(cls, queue_key, connection=None, namespace=None):
         """Returns a Queue instance, based on the naming conventions for naming
         the internal Redis keys.  Can be used to reverse-lookup Queues by their
         Redis keys.
         """
-        prefix = cls.redis_queue_namespace_prefix
+        prefix = build_key(cls.redis_queue_namespace_prefix, namespace)
+
         if not queue_key.startswith(prefix):
             raise ValueError('Not a valid RQ queue key: {0}'.format(queue_key))
         name = queue_key[len(prefix):]
         return cls(name, connection=connection)
```

Comment:
There should be a namespace=namespace positional argument to create an instance with namespace.